internal/rest/client.go

```diff
@@ -334,7 +335,7 @@ func (c *Client) LastConn() time.Time {
 func (c *Client) LastError() error {
 	c.RLock()
 	defer c.RUnlock()
-	return c.lastErr
+	return fmt.Errorf("[%s] %w", c.lastErrTime.Format(time.RFC3339), c.lastErr)
```

Comment:
Hmm this may break things where we don't use `errors.Is()` and there may be places where we don't use it. As this error is encapsulated into NetworkError

My suggestion is don't add it here instead add it as `LastError() (error, time.Time)`  or since no one is consuming `lastErrTime` just record it and leave it for now.

Reply:
I will change it

Reply:
Yeah but NetworkError is not implemented in this manner. So it can lead to other side affects without further testing.

Because we don't know really what it might break.